pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -142,6 +154,67 @@ func (e *cloudWatchExecutor) getRGTAClient(region string) (resourcegroupstagging
 	return e.rgtaClient, nil
 }
 
+func (e *cloudWatchExecutor) getServiceQuotasClient(region string) (servicequotasiface.ServiceQuotasAPI, error) {
+	sess, err := e.newSession(region)
+	if err != nil {
+		return nil, err
+	}
+
+	return newQuotasClient(sess), nil
+}
+
+func (e *cloudWatchExecutor) getQueue(region string) (chan bool, error) {
+	e.queueLock.Lock()
+	defer e.queueLock.Unlock()
+
+	if queue, ok := e.queuesByRegion[region]; ok {
+		return queue, nil
+	}
+
+	concurrentQueriesQuota, err := e.fetchConcurrentQueriesQuota(region)
+	if err != nil {
+		plog.Info("Could not fetch quota")
+	}
+
+	queueChannel := make(chan bool, concurrentQueriesQuota)
+	e.queuesByRegion[region] = queueChannel
+
+	return queueChannel, nil
+}
+
+func (e *cloudWatchExecutor) fetchConcurrentQueriesQuota(region string) (int, error) {
+	client, err := e.getServiceQuotasClient(region)
+	if err != nil {
+		return defaultConcurrentQueries, err
+	}
+
+	concurrentQueriesQuota, err := client.GetServiceQuota(&servicequotas.GetServiceQuotaInput{
+		ServiceCode: aws.String("logs"),
+		QuotaCode:   aws.String("L-32C48FBB"),
+	})
+	if err != nil {
+		return defaultConcurrentQueries, err
+	}
+
+	if concurrentQueriesQuota != nil && concurrentQueriesQuota.Quota != nil {
+		return int(*concurrentQueriesQuota.Quota.Value), nil
+	}
+
+	defaultConcurrentQueriesQuota, err := client.GetAWSDefaultServiceQuota(&servicequotas.GetAWSDefaultServiceQuotaInput{
+		ServiceCode: aws.String("logs"),
+		QuotaCode:   aws.String("L-32C48FBB"),
+	})
+	if err != nil {
+		return defaultConcurrentQueries, err
+	}
+
+	if defaultConcurrentQueriesQuota != nil && defaultConcurrentQueriesQuota.Quota != nil {
+		return int(*defaultConcurrentQueriesQuota.Quota.Value), nil
+	}
+
+	return defaultConcurrentQueries, nil
+}
```

Comment:
After playing with the code locally, I think it makes more sense for this function to not return an error, and instead log a warning about what exactly went wrong. The returned error will only be logged anyhow, so it's pointless.

```suggestion
func (e *cloudWatchExecutor) fetchConcurrentQueriesQuota(region string) int {
	client, err := e.getServiceQuotasClient(region)
	if err != nil {
		plog.Warn("Could not get service quota client")
		return defaultConcurrentQueries
	}

	cqq, err := client.GetServiceQuota(&servicequotas.GetServiceQuotaInput{
		ServiceCode: aws.String("logs"),
		QuotaCode:   aws.String("L-32C48FBB"),
	})
	if err != nil {
		plog.Warn("Could not get service quota")
		return defaultConcurrentQueries
	}

	if cqq != nil && cqq.Quota != nil && cqq.Quota.Value != nil {
		return int(*cqq.Quota.Value)
	}

	plog.Warn("Could not get service quota")

	dcqq, err := client.GetAWSDefaultServiceQuota(&servicequotas.GetAWSDefaultServiceQuotaInput{
		ServiceCode: aws.String("logs"),
		QuotaCode:   aws.String("L-32C48FBB"),
	})
	if err != nil {
		plog.Warn("Could not get default service quota")
		return defaultConcurrentQueries
	}

	if defaultConcurrentQueriesQuota != nil && defaultConcurrentQueriesQuota.Quota != nil {
		return int(*defaultConcurrentQueriesQuota.Quota.Value)
	}

	plog.Warn("Could not get default service quota")
	return defaultConcurrentQueries
}
```